src/components/Recipes/AddToMealplanContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { AddRecipeTitleToMealplanToggle } from '../Button'
+import imagePathPlus from '../../images/inverted-plus.svg'
+import { v4 as uuidv4 } from 'uuid'
+
+import styled from 'styled-components'
+import { saveToLocal, loadFromLocal } from '../../utils/localStorage'
+
+export default function AddToMealplanContainer({ title }) {
+  const [active, setActive] = useState(() => {
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+
+    const initialState =
+      mealplanList.items.find((meal) => {
+        return meal.item === title
+      }) !== undefined
+    return initialState
+  })
+
+  const addRecipeToMealplan = (title) => {
```

Comment:
Diese Funktion würde ich auch auslagern. Ich weiß noch gut von meiner Code Review mit Andreas, dass eine Komponente schon genug zu tun hat, dass sie den State managed und das renderen und sie nicht noch zusätzlich Logik drin haben sollte :)

Reply:
Ich habe mich jetzt dazu entschieden die loadFromLocal Geschichte (die in der Datei 3x vorkommt) in eine loadMealplanList utils function auszulagern und die anderen für jetzt erstmal so zu lassen. In meiner nächsten Story werde ich mein Backend aufsetzten und da muss ich das alles sowieso nochmal umschreiben.